JavaScriptSPA/api.js

```diff
@@ -0,0 +1,47 @@
+// helper function to access the resource with the token
+function callApiWithAccessToken(endpoint, token) {
+    const headers = new Headers();
+    const bearer = `Bearer ${token}`;
+  
+    headers.append("Authorization", bearer);
+  
+    const options = {
+        method: "GET",
+        headers: headers
+      };
+  
+    
+    fetch(endpoint, options)
+      .then(response => response.json())
+      .then(response => {
+        logMessage("Web API returned:\n" + JSON.stringify(response));
+      }).catch(error => {
+        logMessage("Error calling the Web api:\n" + error);
+      });
+}
+
+// calls the resource API with the token
+function callApi() {
+  getTokenPopup(tokenRequest)
```

Comment:
If we are getting thew token here and also accessing the resource, how is it resource_access code only? Can we have some glue code that passes the acquired token to the api.js?

Reply:
makes sense. actually moving this bit to authPopup.js and its redirect version to authRedirect.js made it more convenient. Now the user just have to switch the files to switch the flows.